src/test/java/com/devonfw/sample/archunit/ComponentRuleTest.java

```diff
@@ -0,0 +1,328 @@
+      isAllowed = true;
+    }
+    // Components may always depend on the default business component.
+    if (targetPkg.isComponentGeneral() && (targetPkg.isLayerLogic() || targetPkg.isLayerCommon())) {
+      isAllowed = true;
+    }
+    return isAllowed;
+  }
+
+  /**
+   * Dependency of a components dataaccess layer towards the same components common layer is allowed. In addition a
+   * dependency towards the projects default component is allowed too.
+   */
+  private static boolean isComponentDataaccessLayerCompliant(PackageStructure sourcePkg, PackageStructure targetPkg) {
+
+    boolean isAllowed = false;
+    // Components dataaccess layer can depend on their own dataaccess and common layer.
+    if (sourcePkg.hasSameComponent(targetPkg) && (targetPkg.isLayerDataAccess() || targetPkg.isLayerCommon())) {
+      isAllowed = true;
+    }
+    // Components may always depend on the default business component.
+    if (targetPkg.isComponentGeneral() && (targetPkg.isLayerDataAccess() || targetPkg.isLayerCommon())) {
+      isAllowed = true;
+    }
+    return isAllowed;
+  }
+
+  /**
+   * Dependency of a components logic layer towards the same components dataaccess or common layer is allowed. In
+   * addition a dependency towards the projects default component is allowed too.
+   */
+  private static boolean isComponentLogicLayerCompliant(PackageStructure sourcePkg, PackageStructure targetPkg) {
+
+    boolean isAllowed = false;
+    // Components logic layer can depend on their own lower layers: dataaccess and
+    // common.
+    if (sourcePkg.hasSameComponent(targetPkg) && (targetPkg.isLayerDataAccess() || targetPkg.isLayerCommon())) {
+      isAllowed = true;
+    }
+    // Components may always depend on the default business component.
+    if (targetPkg.isComponentGeneral() && (targetPkg.isLayerDataAccess() || targetPkg.isLayerCommon())) {
+      isAllowed = true;
+    }
+    return isAllowed;
+  }
+
+  /**
+   * Dependency of a components service layer towards the same components logic or common layer is allowed. In addition
+   * a dependency towards the projects default component is allowed too.
+   *
+   * @param sourceClass Source JavaClass to check if dependencies from itself towards targetClass are allowed.
+   * @param targetClass Target JavaClass to check if dependencies from sourceClass towards it are allowed.
+   */
+  private static boolean isComponentServiceLayerCompliantTowardsComponentsLogicLayer(PackageStructure sourcePkg,
+      PackageStructure targetPkg) {
+
+    boolean isAllowed = false;
+    // Components service layer can depend on their own lower layers: logic and
+    // common.
+    if (sourcePkg.hasSameComponent(targetPkg) && (targetPkg.isLayerLogic() || targetPkg.isLayerCommon())) {
+      isAllowed = true;
+    }
+    // Components may always depend on the default business component.
+    if (targetPkg.isComponentGeneral() && (targetPkg.isLayerLogic() || targetPkg.isLayerCommon())) {
```

Comment:
I have added an ArchRule that checks whether the `general` component does not depend on any other component.

For 1: I have changed the tests to check for disallowed dependencies rather than what is allowed.